pkg/logs/decoder/decoder.go

```diff
@@ -91,7 +91,19 @@ func NewDecoderWithEndLineMatcher(source *config.LogSource, parser parser.Parser
 
 	for _, rule := range source.Config.ProcessingRules {
 		if rule.Type == config.MultiLine {
-			lineHandler = NewMultiLineHandler(outputChan, rule.Regex, config.AggregationTimeout(), lineLimit)
+			lh := NewMultiLineHandler(outputChan, rule.Regex, config.AggregationTimeout(), lineLimit)
+
+			// Since a single source can have multiple file tailers - each with their own decoder instance,
+			// Make sure we keep track of the multiline match count info from all of the decoders so the
+			// status page displays it correctly.
+			if existingInfo, ok := source.GetInfo(lh.countInfo.InfoKey()).(*config.CountInfo); ok {
```

Comment:
`GetInfo` is used to check if the key exists. What do you think about replacing `GetInfo` by `HasKeyInfo(key string) bool`?

Reply:
two things are happening here: 
1. looking up an existing info provider
2. downcast it to a `CountInfo`

I am relying on the fact that a type assertion on a `nil` reference returns `ok` as `false` instead of crashing (or having to do an extra check). With this in mind, what do you think?